CoreCLR failed to build due to error MSB1008 with MSVC on windows

Environment: Windows Server 2016 + VS2017 CoreCLR master branch latest srouce code

We tried to build CoreCLR and it failed to build due to error MSB1008: Only one project can be specified.  It can be first reproduced on 929da85 revision. It should be a CoreCLR source issue. Could you look at this issue? Thanks a lot.

Repro steps:
1.git clone https://github.com/dotnet/coreclr D:\CoreCLR\src
2.Open a clean x86 prompt and browse to D:\CoreCLR\src
3.build x86 release -- /clp:ShowCommandLine

[log_x86_build.log](https://github.com/dotnet/coreclr/files/3476195/log_x86_build.log)

Error info:
Time Elapsed 00:01:10.28
BUILDTEST: Starting the Managed Tests Build
BUILDTEST: Building tests divided into 3 test groups
Running: msbuild D:\CoreCLR\src\tests\build.proj /flp:Verbosity=normal;LogFile="D:\CoreCLR\src\bin\Logs\Tests_Managed_Windows_NT__x86__Release.log";Append=false /flp1:WarningsOnly;LogFile="D:\CoreCLR\src\bin\Logs\Tests_Managed_Windows_NT__x86__Release.wrn";Append=false /flp2:ErrorsOnly;LogFile="D:\CoreCLR\src\bin\Logs\Tests_Managed_Windows_NT__x86__Release.err";Append=false /p:TargetsWindows=true /p:__BuildOS=Windows_NT /p:__BuildType=Release /p:__BuildArch=x86 /nologo /verbosity:minimal /clp:Summary /maxcpucount     /clp:ShowCommandLine
MSBUILD : error MSB1008: Only one project can be specified.
Switch:  /clp:ShowCommandLine

For switch syntax, type "MSBuild -help"
Build failed.
BUILDTEST: Error: managed test build failed. Refer to the build log files for details:
    D:\CoreCLR\src\bin\Logs\Tests_Managed_Windows_NT__x86__Release.log
    D:\CoreCLR\src\bin\Logs\Tests_Managed_Windows_NT__x86__Release.wrn
    D:\CoreCLR\src\bin\Logs\Tests_Managed_Windows_NT__x86__Release.err
[command took 1076 seconds]

--- a/build-test.cmd
+++ b/build-test.cmd
@@ -273,14 +273,14 @@ set __BuildErr=%__LogsDir%\%__BuildLogRootName%_%__BuildOS%__%__BuildArch%__%__B
 set __MsbuildLog=/flp:Verbosity=normal;LogFile="%__BuildLog%"
 set __MsbuildWrn=/flp1:WarningsOnly;LogFile="%__BuildWrn%"
 set __MsbuildErr=/flp2:ErrorsOnly;LogFile="%__BuildErr%"
-set __Logging=!__MsbuildLog! !__MsbuildWrn! !__MsbuildErr!
+set __Logging='!__MsbuildLog!' '!__MsbuildWrn!' '!__MsbuildErr!'
 
 REM Disable warnAsError - coreclr issue 19922
 powershell -NoProfile -ExecutionPolicy ByPass -NoLogo -Command "%__ProjectDir%\eng\common\msbuild.ps1" %__ArcadeScriptArgs%^
   %__ProjectDir%\tests\build.proj -warnAsError:0 /t:BatchRestorePackages /nodeReuse:false^
   /p:RestoreDefaultOptimizationDataPackage=false /p:PortableBuild=true^
   /p:UsePartialNGENOptimization=false /maxcpucount^
-  '!__Logging!' %__CommonMSBuildArgs% %__PriorityArg% %__UnprocessedBuildArgs%
+  !__Logging! %__CommonMSBuildArgs% %__PriorityArg% %__UnprocessedBuildArgs%
 
 :SkipRestoreProduct
 
@@ -319,14 +319,14 @@ for /l %%G in (1, 1, %__NumberOfTestGroups%) do (
     set __MsbuildLog=/flp:Verbosity=normal;LogFile="%__BuildLog%";Append=!__AppendToLog!
     set __MsbuildWrn=/flp1:WarningsOnly;LogFile="%__BuildWrn%";Append=!__AppendToLog!
     set __MsbuildErr=/flp2:ErrorsOnly;LogFile="%__BuildErr%";Append=!__AppendToLog!
-    set __Logging=!__MsbuildLog! !__MsbuildWrn! !__MsbuildErr!
+    set __Logging='!__MsbuildLog!' '!__MsbuildWrn!' '!__MsbuildErr!'
 
     set __TestGroupToBuild=%%G
     echo Running: msbuild %__ProjectDir%\tests\build.proj !__MsbuildLog! !__MsbuildWrn! !__MsbuildErr! %TargetsWindowsMsbuildArg% %__msbuildArgs% !__PriorityArg! %__UnprocessedBuildArgs%
 
     REM Disable warnAsError - coreclr issue 19922
     powershell -NoProfile -ExecutionPolicy ByPass -NoLogo -Command "%__ProjectDir%\eng\common\msbuild.ps1" %__ArcadeScriptArgs%^
-        %__ProjectDir%\tests\build.proj -warnAsError:0 '!__Logging!' %TargetsWindowsMsbuildArg% %__msbuildArgs% !__PriorityArg! '%__UnprocessedBuildArgs%'
+        %__ProjectDir%\tests\build.proj -warnAsError:0 !__Logging! %TargetsWindowsMsbuildArg% %__msbuildArgs% !__PriorityArg! %__UnprocessedBuildArgs%
 
     if errorlevel 1 (
         echo %__ErrMsgPrefix%%__MsgPrefix%Error: managed test build failed. Refer to the build log files for details:

--- a/build.cmd
+++ b/build.cmd
@@ -649,7 +649,7 @@ if %__BuildCoreLib% EQU 1 (
         powershell -NoProfile -ExecutionPolicy ByPass -NoLogo -Command "%__ProjectDir%\eng\common\msbuild.ps1" %__ArcadeScriptArgs%^
             %__ProjectDir%\src\build.proj -warnAsError:0^
             /nodeReuse:false /p:PortableBuild=true /maxcpucount /p:ArcadeBuild=true^
-            '!__Logging!' %__CommonMSBuildArgs% !__ExtraBuildArgs! %__UnprocessedBuildArgs%
+            '!__MsbuildLog!' '!__MsbuildWrn!' '!__MsbuildErr!' %__CommonMSBuildArgs% !__ExtraBuildArgs! %__UnprocessedBuildArgs%
         if not !errorlevel! == 0 (
             echo %__ErrMsgPrefix%%__MsgPrefix%Error: Managed Product assemblies build failed. Refer to the build log files for details.
             echo     !__BuildLog!

